core/block/object/systemobjectupdate/updater.go

```diff
@@ -0,0 +1,198 @@
+package systemobjectupdate
+
+import (
+	"context"
+
+	"github.com/anyproto/any-sync/app"
+	"github.com/anyproto/any-sync/commonspace/spacestorage"
+	"github.com/gogo/protobuf/types"
+	"github.com/samber/lo"
+
+	"github.com/anyproto/anytype-heart/core/block/editor/basic"
+	"github.com/anyproto/anytype-heart/core/block/getblock"
+	"github.com/anyproto/anytype-heart/core/domain"
+	"github.com/anyproto/anytype-heart/core/relationutils"
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	"github.com/anyproto/anytype-heart/pkg/lib/database"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/addr"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+	"github.com/anyproto/anytype-heart/pkg/lib/logging"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+	"github.com/anyproto/anytype-heart/space/spacecore/storage"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
+)
+
+const CName = "systemobjectupdater"
+
+var log = logging.Logger("system-objects-updater")
+
+type SystemObjectUpdater struct {
+	app.ComponentRunnable
+
+	store   objectstore.ObjectStore
+	storage storage.ClientStorage
+	picker  getblock.ObjectGetter
+}
+
+func New() *SystemObjectUpdater {
+	return &SystemObjectUpdater{}
+}
+
+func (u *SystemObjectUpdater) Init(a *app.App) error {
+	u.store = a.MustComponent(objectstore.CName).(objectstore.ObjectStore)
+	u.storage = a.MustComponent(spacestorage.CName).(storage.ClientStorage)
+	u.picker = app.MustComponent[getblock.ObjectGetter](a)
+	return nil
+}
+
+func (u *SystemObjectUpdater) Name() string {
+	return CName
+}
+
+func (u *SystemObjectUpdater) Run(_ context.Context) error {
+	go u.updateSystemObjects()
+	return nil
+}
+
+func (u *SystemObjectUpdater) Close(_ context.Context) error {
+	return nil
+}
+
+func (u *SystemObjectUpdater) updateSystemObjects() {
+	marketRels, err := u.store.ListAllRelations(addr.AnytypeMarketplaceWorkspace)
+	if err != nil {
+		log.Errorf("failed to get relations from marketplace space")
```

Comment:
Why don't we include err in logging? The same question about line 71 and 77

Reply:
Thank you! Fixed